spyder/plugins/editor/extensions/tests/test_docstring.py

```diff
@@ -0,0 +1,355 @@
+  (List[Tuple[str, float]], str, float):
+    """\n    \n
+    Parameters
+    ----------
+    arg : TYPE
+        DESCRIPTION
+    arg0 : TYPE
+        DESCRIPTION
+    arg1 : int
+        DESCRIPTION
+    arg2 : List[Tuple[str, float]]
+        DESCRIPTION
+    arg3 : TYPE, optional
+        DESCRIPTION (the default is '-> (float, int):')
+    arg4 : TYPE, optional
+        DESCRIPTION (the default is ':float, int[')
+    arg5 : str, optional
+        DESCRIPTION (the default is '""')
+
+    Returns
+    -------
+    (List[Tuple[str, float]], str, float)
+        DESCRIPTION
+
+    """
+    '''),
+        ('Googledoc',
+         '''async def foo():
+    ''',
+         '''async def foo():
+    """\n    \n
+    Returns:
+        RETURN_TYPE: DESCRIPTION
+
+    """
+    '''
+         ),
+        ('Googledoc',
+         '''  def foo():
+      ''',
+         '''  def foo():
+      """\n      \n
+      Returns:
+          RETURN_TYPE: DESCRIPTION
+
+      """
+      ''',
+         ),
+        ('Googledoc',
+         '''def foo(arg, arg0, arg1: int, arg2: List[Tuple[str, float]],
+    arg3='-> (float, int):', arg4=':float, int[', arg5: str='""') -> \
+  (List[Tuple[str, float]], str, float):
+    ''',
+         '''def foo(arg, arg0, arg1: int, arg2: List[Tuple[str, float]],
+    arg3='-> (float, int):', arg4=':float, int[', arg5: str='""') -> \
+  (List[Tuple[str, float]], str, float):
+    """\n    \n
+    Args:
+        arg (TYPE): DESCRIPTION
+        arg0 (TYPE): DESCRIPTION
+        arg1 (int): DESCRIPTION
+        arg2 (List[Tuple[str, float]]): DESCRIPTION
+        arg3 (TYPE, optional): Defaults to '-> (float, int):'. DESCRIPTION
+        arg4 (TYPE, optional): Defaults to ':float, int['. DESCRIPTION
+        arg5 (str, optional): Defaults to '""'. DESCRIPTION
+
+    Returns:
+        (List[Tuple[str, float]], str, float): DESCRIPTION
+
+    """
+    '''),
+    ])
+def test_editor_docstring_by_shortcut(qtbot, editor_auto_docstring, doc_type,
+                                      text, expected, use_shortcut):
+    """Test auto docstring by shortcut."""
+    CONF.set('editor', 'docstring_type', doc_type)
+    editor = editor_auto_docstring
+    editor.set_text(text)
+
+    cursor = editor.textCursor()
+    cursor.setPosition(0, QTextCursor.MoveAnchor)
+    editor.setTextCursor(cursor)
+    writer = editor.writer_docstring
+
+    if use_shortcut:
+        writer.write_docstring_for_shortcut()
+    else:
+        pos = editor.cursorRect().bottomRight()
+        pos = editor.mapToGlobal(pos)
+        writer.line_number_cursor = editor.get_line_number_at(pos)
+        writer.write_docstring_at_first_line_of_function()
+
+    assert editor.toPlainText() == expected
+
+
+# =============================================================================
+# ---- Example for manual Testing of numpy docstring
+# =============================================================================
```

Comment:
I appreciate the effort, but this really should not be in the test file, but rather in a separate one, if present at all.